hash.go

```diff
@@ -40,8 +40,18 @@ func (h Hash) String() string {
 }
 
 // Sum builds a digest according to the specs in the Holochain
-func (h *Hash) Sum(hc HashSpec, data []byte) (err error) {
-	h.H, err = mh.Sum(data, hc.Code, hc.Length)
+func (hash *Hash) Sum(holo *Holochain, data []byte) (err error) {
```

Comment:
func (holo *Holochain) Sum() (hash Hash, err error) { ...

For all similar patterns.